metricflow/test/model/validations/test_unique_valid_name.py

```diff
@@ -13,6 +16,10 @@ def copied_model(simple_model__pre_transforms: UserConfiguredModel) -> UserConfi
     return copy.deepcopy(simple_model__pre_transforms)
 
 
+def _categorical_dimensions(data_source: DataSource) -> Tuple[Dimension, ...]:
```

Comment:
move to `test/test_utils.py`

Reply:
I prefer to leave this where it is and open a separate issue to rewrite this whole test to use purpose-built broken models instead of doing all of this mutation. As it is currently written the test is exceedingly difficult to understand, and for all I know there are more of these non-deterministic code gremlins lurking in the shadows and we've just been lucky.